core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -136,7 +136,8 @@ private[spark] class CoarseMesosSchedulerBackend(
       sc.sparkUser,
       sc.appName,
       sc.conf,
-      sc.ui.map(_.appUIAddress))
+      sc.conf.getOption("spark.mesos.webui.url").orElse(sc.ui.map(_.appUIAddress))
```

Comment:
well, `sc.ui.map(...)` is an Option construct itself, thus this construct. `orElse` accepts an Option like `flatMap`, whereas `.get(val, default)` expects an explicit value.